packages/auth/src/core/credentials/email.ts

```diff
@@ -166,7 +169,7 @@ export class EmailAuthCredential extends AuthCredential {
   ): Promise<IdTokenResponse> {
     switch (this.signInMethod) {
       case SignInMethod.EMAIL_PASSWORD:
-        return updateEmailPassword(auth, {
+        return linkEmailPassword(auth, {
```

Comment:
Once https://github.com/firebase/firebase-js-sdk/pull/7666 merges, we should also wrap this with reCAPTCHA Enterprise logic.